src/lib/statistics/column_statistics.cpp

```diff
@@ -2,9 +2,46 @@
 
 #include <sstream>
 
+#include "lossless_cast.hpp"
 #include "resolve_type.hpp"
 #include "table_statistics.hpp"
-#include "type_cast.hpp"
+
+namespace {
+
+using namespace opossum;  // NOLINT
+
+// Statistics - being estimations - are one of the few places where we are okay with some information loss and therefore
+// do not use lossless_cast
+template <typename Target>
+std::optional<Target> static_variant_cast(const AllTypeVariant& source) {
+  if (variant_is_null(source)) return std::nullopt;
+
+  std::optional<Target> result;
+
+  resolve_data_type(data_type_from_all_type_variant(source), [&](const auto source_data_type_t) {
+    using SourceDataType = typename decltype(source_data_type_t)::type;
+
+    if constexpr (std::is_same_v<Target, SourceDataType>) {
+      result = boost::get<SourceDataType>(source);
+    } else {
+      if constexpr (std::is_same_v<pmr_string, SourceDataType> == std::is_same_v<pmr_string, Target>) {
+        const auto source_value = boost::get<SourceDataType>(source);
+        if (source_value <= std::numeric_limits<Target>::max() &&
+            source_value >= std::numeric_limits<Target>::lowest()) {
+          result = static_cast<Target>(boost::get<SourceDataType>(source));
+        } else {
+          std::cout << "Fuck" << std::endl;
```

Comment:
;)

Either use those max values or return empty statistics.

Reply:
Sorry. Doing hacky fixes before my first coffee....

Reply:
empty statistics it is

Reply:
changed my mind and went for min/max values